### PR TITLE
Use osmosis-rust as finschia-wasm

### PR DESCRIPTION
**Pull request by kokeshiM0chi**
# Description

To use osmosis-rust as finschia-wasm, change the submodule and package name.

## changes

This PR has the following changes:
- Switch `dependencies/` and `.gitmodules` to submodule for finschia (Along with this, the tag specified in proto-build's main.rs has been changed to the finschia repository tag.)
- Change package and variable names derived from osmosis to those of finschia.

## remaining issues

These were deleted in order to consider the ones for finschia from scratch.
- examples/
- tests/

This is commented out because it is necessary to replace it with a typical message under `finschia-std/types` and rewrite the test.
- packages/finschia-std-derive/tests

https://github.com/Finschia/finschia-wasm/issues/2

### CODE REVIEW

**Comment by loloicci** [comment]
Can't you rewrite tests in [c1c2d65](https://github.com/Finschia/finschia-wasm/pull/1/commits/c1c2d6579e0a82069baa9e959701c1febe8ee59f) rather than delete them?

**Comment by kokeshiM0chi** [comment]
@loloicci Thanks. I added tests for finschia. PTAL

File: Cargo.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [workspace]
 resolver = "2"
-members = ["packages/*", "tests/*"]
+members = ["packages/*"]
```
Comment (loloicci) [inline Cargo.toml:3]: Why "tests/*" is deleted?

Comment (kokeshiM0chi) [reply Cargo.toml:3]: This tests folder is originally an integration test using contracts. Therefore, consider integration tests when adding examples/contracts.

**Review by loloicci — APPROVED** [review]

**Pull request merged**